signer/migrations/0003__create_tables.sql

```diff
@@ -184,3 +177,81 @@ CREATE TABLE sbtc_signer.withdrawal_reject_events (
+-- the individual Bitcoin transactions they are related to.
+CREATE TABLE sbtc_signer.sweep_transactions (
+    -- The Bitcoin transaction ID of the transaction.
+    txid BYTEA PRIMARY KEY NOT NULL,
+    -- The signer UTXO being spent in this transaction.
+    signer_prevout_txid BYTEA NOT NULL,
+    signer_prevout_output_index INTEGER NOT NULL,
+    signer_prevout_amount BIGINT NOT NULL,
+    signer_prevout_script_pubkey BYTEA NOT NULL,
+    -- The total amount of the transaction.
+    amount BIGINT NOT NULL,
+    -- The fee paid for the transaction.
+    fee BIGINT NOT NULL,
+    -- The Bitcoin block hash at which this package was created.
+    created_at_block_hash BYTEA NOT NULL,
+    -- The Bitcoin market fee rate at the time this package was created.
+    market_fee_rate DOUBLE PRECISION NOT NULL,
+    -- Timestamp of when this package was created.
+    created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+-- Represents a single withdrawal request which has been included in a sweep
+-- transaction package. Withdrawal requests have a unique ID so we use that here
+-- to reference the withdrawal request together with its Stacks block hash,
+-- which can be retrieved from the `withdrawal_requests` table.
+CREATE TABLE sbtc_signer.swept_withdrawals (
+    sweep_transaction_txid BYTEA PRIMARY KEY NOT NULL,
+    -- The index of the sweep output in the sweep transaction.
+    output_index INTEGER NOT NULL,
+    -- The ID of the withdrawal request, referencing the `withdrawal_requests`
+    -- table.
+    withdrawal_request_id BIGINT NOT NULL,
+    -- The Stacks block hash of the withdrawal request, referencing the
+    -- `withdrawal_requests` table.
+    withdrawal_request_block_hash BYTEA NOT NULL,
+
+    FOREIGN KEY (sweep_transaction_txid) 
+        REFERENCES sbtc_signer.sweep_transactions(txid),
+
+    FOREIGN KEY (withdrawal_request_id, withdrawal_request_block_hash) 
+        REFERENCES sbtc_signer.withdrawal_requests(request_id, block_hash)
+);
+-- Our main index which will cover searches by 'withdrawal_request_id' and
+-- 'withdrawal_request_block_hash' while also restricting the combination to be
+-- unique per 'sweep_transaction_id'.
+CREATE UNIQUE INDEX uix_swept_req_id_req_block_hash_pkgd_txid 
+    ON sbtc_signer.swept_withdrawals(withdrawal_request_id, withdrawal_request_block_hash, sweep_transaction_txid);
+
+-- Represents a single deposit request which has been included in a
+-- transaction package. Deposit requests do not have a unique ID in the same way
+-- as withdrawal requests, so we reference the Bitcoin transaction ID and output
+-- index instead, which can be retrieved from the `deposit_requests` table.
+CREATE TABLE sbtc_signer.swept_deposits (
+    -- References the `packaged_transaction` in which this deposit was included.
+    sweep_transaction_txid BYTEA NOT NULL,
+    -- The index of the sweep input in the sweep transaction.
+    input_index INTEGER NOT NULL,
```

Comment:
These two could be the primary key here. Or did you not want to have an explicit primary key for this table?

Reply:
Good catch, just a miss

Reply:
[a0bac87](https://github.com/stacks-network/sbtc/pull/710/commits/a0bac8711f4e83727a72f11d7a147633d8df4bd6)